Common/Tools/TrackTuner.h

```diff
@@ -476,38 +493,72 @@ struct TrackTuner : o2::framework::ConfigurableGroup {
     }
+    for (int iphibin = 0; iphibin < nPhiBins; ++iphibin) {
 
-    std::string grDcaZResNameMC = "resCurrentDcaZ";
-    std::string grDcaZMeanNameMC = "meanCurrentDcaZ";
-    std::string grDcaZPullNameMC = "pullsCurrentDcaZ";
-    std::string grDcaZResNameData = "resUpgrDcaZ";
-    std::string grDcaZMeanNameData = "meanUpgrDcaZ";
-    std::string grDcaZPullNameData = "pullsUpgrDcaZ";
-
-    grDcaZResVsPtPionMC.reset(dynamic_cast<TGraphErrors*>(td->Get(grDcaZResNameMC.c_str())));
-    grDcaZResVsPtPionData.reset(dynamic_cast<TGraphErrors*>(td->Get(grDcaZResNameData.c_str())));
-    grDcaZMeanVsPtPionMC.reset(dynamic_cast<TGraphErrors*>(td->Get(grDcaZMeanNameMC.c_str())));
-    grDcaZMeanVsPtPionData.reset(dynamic_cast<TGraphErrors*>(td->Get(grDcaZMeanNameData.c_str())));
-    grDcaZPullVsPtPionMC.reset(dynamic_cast<TGraphErrors*>(td->Get(grDcaZPullNameMC.c_str())));
-    grDcaZPullVsPtPionData.reset(dynamic_cast<TGraphErrors*>(td->Get(grDcaZPullNameData.c_str())));
-    if (!grDcaZResVsPtPionMC.get() || !grDcaZResVsPtPionData.get() || !grDcaZMeanVsPtPionMC.get() || !grDcaZMeanVsPtPionData.get() || !grDcaZPullVsPtPionMC.get() || !grDcaZPullVsPtPionData.get()) {
-      LOG(fatal) << "Something wrong with the names of the correction graphs for dcaZ. Fix it!";
+      grDcaXYResVsPtPionMC[iphibin].reset(loadGraph(iphibin, "resCurrentDcaXY").release());
+      grDcaXYResVsPtPionData[iphibin].reset(loadGraph(iphibin, "resUpgrDcaXY").release());
+      grDcaXYMeanVsPtPionMC[iphibin].reset(loadGraph(iphibin, "meanCurrentDcaXY").release());
+      grDcaXYMeanVsPtPionData[iphibin].reset(loadGraph(iphibin, "meanUpgrDcaXY").release());
+      grDcaXYPullVsPtPionMC[iphibin].reset(loadGraph(iphibin, "pullsCurrentDcaXY").release());
+      grDcaXYPullVsPtPionData[iphibin].reset(loadGraph(iphibin, "pullsUpgrDcaXY").release());
+
+      if (!grDcaXYResVsPtPionMC[iphibin].get() || !grDcaXYResVsPtPionData[iphibin].get() || !grDcaXYMeanVsPtPionMC[iphibin].get() || !grDcaXYMeanVsPtPionData[iphibin].get() || !grDcaXYPullVsPtPionMC[iphibin].get() || !grDcaXYPullVsPtPionData[iphibin].get()) {
+        LOG(fatal) << "[TrackTuner]     Something wrong with the names of the correction graphs for dcaXY. Fix it!";
+      }
+    }
+
+    if (inputNphiBins != 0) {
+      LOG(info) << "[TrackTuner]    Loading phi-dependent Z TGraphErrors";
+    }
+    for (int iphibin = 0; iphibin < nPhiBins; ++iphibin) {
+      grDcaZResVsPtPionMC[iphibin].reset(loadGraph(iphibin, "resCurrentDcaZ").release());
+      grDcaZMeanVsPtPionMC[iphibin].reset(loadGraph(iphibin, "meanCurrentDcaZ").release());
+      grDcaZPullVsPtPionMC[iphibin].reset(loadGraph(iphibin, "pullsCurrentDcaZ").release());
+      grDcaZResVsPtPionData[iphibin].reset(loadGraph(iphibin, "resUpgrDcaZ").release());
+      grDcaZMeanVsPtPionData[iphibin].reset(loadGraph(iphibin, "meanUpgrDcaZ").release());
+      grDcaZPullVsPtPionData[iphibin].reset(loadGraph(iphibin, "pullsUpgrDcaZ").release());
+
```

Comment:
same comment as above

```suggestion
    for (int iPhiBin = 0; iPhiBin < nPhiBins; ++iPhiBin) {
      grDcaZResVsPtPionMC[iPhiBin].reset(loadGraph(iPhiBin, "resCurrentDcaZ"));
      grDcaZMeanVsPtPionMC[iPhiBin].reset(loadGraph(iPhiBin, "meanCurrentDcaZ"));
      grDcaZPullVsPtPionMC[iPhiBin].reset(loadGraph(iPhiBin, "pullsCurrentDcaZ"));
      grDcaZResVsPtPionData[iPhiBin].reset(loadGraph(iPhiBin, "resUpgrDcaZ"));
      grDcaZMeanVsPtPionData[iPhiBin].reset(loadGraph(iPhiBin, "meanUpgrDcaZ"));
      grDcaZPullVsPtPionData[iPhiBin].reset(loadGraph(iPhiBin, "pullsUpgrDcaZ"));

```

Reply:
ok!